pkg/apis/internal.admin.acorn.io/v1/resources.go

```diff
@@ -30,44 +37,75 @@ type Resources struct {
 
 // Add will add the resources of another Resources struct into the current one.
 func (current *Resources) Add(incoming Resources) {
-	current.Apps += incoming.Apps
-	current.Containers += incoming.Containers
-	current.Jobs += incoming.Jobs
-	current.Volumes += incoming.Volumes
-	current.Secrets += incoming.Secrets
-	current.Images += incoming.Images
-	current.Projects += incoming.Projects
-
-	current.VolumeStorage.Add(incoming.VolumeStorage)
-	current.Memory.Add(incoming.Memory)
-	current.CPU.Add(incoming.CPU)
+	add := func(c, i int) int {
+		if c == Unlimited || i == Unlimited {
+			return Unlimited
+		}
+		return c + i
+	}
+
+	addQuantity := func(c, i resource.Quantity) resource.Quantity {
+		if c.Equal(UnlimitedQuantity()) || i.Equal(UnlimitedQuantity()) {
+			return UnlimitedQuantity()
+		}
+		c.Add(i)
+		return c
+	}
+
+	current.Apps = add(current.Apps, incoming.Apps)
+	current.Containers = add(current.Containers, incoming.Jobs)
+	current.Jobs = add(current.Jobs, incoming.Jobs)
+	current.Volumes = add(current.Volumes, incoming.Volumes)
+	current.Secrets = add(current.Secrets, incoming.Secrets)
+	current.Images = add(current.Images, incoming.Images)
+	current.Projects = add(current.Projects, incoming.Projects)
+
+	current.VolumeStorage = addQuantity(current.VolumeStorage, incoming.VolumeStorage)
+	current.Memory = addQuantity(current.Memory, incoming.Memory)
+	current.CPU = addQuantity(current.CPU, incoming.CPU)
 }
 
-// Remove will remove the resources of another Resources struct from the current one.
+// Remove will remove the resources of another Resources struct from the current one. Calling remove
+// will be a no-op for any resource values that are set to unlimited.
 func (current *Resources) Remove(incoming Resources, all bool) {
-	// Do not allow resources to go below 0
-	nonNegativeSubtract := func(currentVal, incomingVal int) int {
-		difference := currentVal - incomingVal
+	sub := func(c, i int) int {
+		if c == Unlimited || i == Unlimited {
+			return c
```

Comment:
Should this be
```suggestion
			return Unlimited
```

Reply:
@tylerslaton Just wanted to double-check whether this is correct. Same for the below.

Reply:
Talked about this offline and left a comment to reflect why this code exists and what it is doing.